chainio/txmgr/txmgr.go

```diff
@@ -0,0 +1,240 @@
+	GetNoSendTxOpts() (*bind.TransactOpts, error)
+}
+
+type EthBackend interface {
+	bind.ContractCaller
+	bind.ContractTransactor
+	bind.ContractFilterer
+
+	// TransactionReceipt queries the backend for a receipt associated with
+	// txHash. If lookup does not fail, but the transaction is not found,
+	// nil should be returned for both values.
+	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
+
+	ChainID(ctx context.Context) (*big.Int, error)
+}
+
+type SimpleTxManager struct {
+	backend EthBackend
+
+	signerFn signerv2.SignerFn
+
+	log logging.Logger
+
+	from common.Address
+
+	contracts map[common.Address]*bind.BoundContract
+}
+
+var _ TxManager = (*SimpleTxManager)(nil)
+
+func NewSimpleTxManager(
+	backend EthBackend,
+	log logging.Logger,
+	signerFn signerv2.SignerFn,
+	from common.Address,
+) *SimpleTxManager {
+	return &SimpleTxManager{
+		backend:   backend,
+		log:       log,
+		signerFn:  signerFn,
+		from:      from,
+		contracts: map[common.Address]*bind.BoundContract{},
+	}
+}
+
+func (m *SimpleTxManager) Send(ctx context.Context, tx *types.Transaction) (*types.Receipt, error) {
+
+	// Estimate gas and nonce
+	m.log.Debug("Estimating gas and nonce", "tx", tx.Hash().Hex())
+	tx, err := m.estimateGasAndNonce(ctx, tx)
+	if err != nil {
+		return nil, err
+	}
+
+	m.log.Debug("Getting signer for tx", "tx", tx.Hash().Hex())
+	signer, err := m.signerFn(ctx, m.from)
+	if err != nil {
+		return nil, err
+	}
+
+	m.log.Debug("Sending transaction", "tx", tx.Hash().Hex())
+	opts := &bind.TransactOpts{
+		From:      m.from,
+		Nonce:     new(big.Int).SetUint64(tx.Nonce()),
+		Signer:    signer,
+		Value:     tx.Value(),
+		GasFeeCap: tx.GasFeeCap(),
+		GasTipCap: tx.GasTipCap(),
+		GasLimit:  tx.Gas(),
+		Context:   ctx,
+	}
+
+	contract := m.contracts[*tx.To()]
+	// if the contract has not been cached
+	if contract == nil {
+		// create a dummy bound contract tied to the `to` address of the transaction
+		contract = bind.NewBoundContract(*tx.To(), abi.ABI{}, m.backend, m.backend, m.backend)
+		// cache the contract for later use
+		m.contracts[*tx.To()] = contract
+	}
+
+	tx, err = contract.RawTransact(opts, tx.Data())
+	if err != nil {
+		return nil, fmt.Errorf("send: failed to send txn: %w", err)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	receipt := m.waitForTx(ctx, tx)
+
+	return receipt, nil
+}
+
+// GetNoSendTxOpts This generates a noSend TransactOpts so that we can use
+// this to generate the transaction without actually sending it
+func (m *SimpleTxManager) GetNoSendTxOpts() (*bind.TransactOpts, error) {
+	signer, err := m.signerFn(context.Background(), m.from)
+	if err != nil {
+		return nil, err
+	}
+	noSendTxOpts := &bind.TransactOpts{
+		From:   m.from,
+		Signer: signer,
+		NoSend: true,
+	}
+	return noSendTxOpts, nil
+}
+
+func (m *SimpleTxManager) estimateGasAndNonce(ctx context.Context, tx *types.Transaction) (*types.Transaction, error) {
```

Comment:
Unclear to me how this is different from what the default geth bindings are doing. See https://github.com/ethereum/go-ethereum/blob/4d9f3cd5d751efccd501b08ab6cf38a83b5e2858/accounts/abi/bind/base.go#L292

Can we add comment above to explain what exactly is different, aka why all this code is needed.

Reply:
yes. two reasons
- We are supporting legacy ETH clients who doesn't have `SuggestGasTipCap` imoplemented.
- We can manipulate gas easily by adding buffer. I have a TODO here - https://github.com/Layr-Labs/eigensdk-go/pull/75/files#diff-223616ebe949cff704b8d355a461951d12ccf06b07d593d9983879d1735bf140R186 


will add these reasons.

Reply:
added reasons as comments.